rclcpp/src/rclcpp/executor.cpp

```diff
@@ -22,12 +22,21 @@ using rclcpp::executor::Executor;
 
 Executor::Executor(rclcpp::memory_strategy::MemoryStrategy::SharedPtr ms)
 : spinning(false), interrupt_guard_condition_(rmw_create_guard_condition()),
+  waitset_(rmw_create_waitset(0)),
```

Comment:
rmw_create_guard_condition can also return nullptr.

I will change the constructor to check both values and throw an error if it failed.